src/chainstate/stacks/miner.rs

```diff
@@ -3994,6 +4006,131 @@ pub mod test {
         }
     }
 
+    #[test]
```

Comment:
From what I can tell, there's not a way to automatically test that the miner is being silent -- you'd have to look at the test output to determine this, right?

Reply:
correct.  One thing that we may consider, is to temporarily redirect `stdout` into a buffer, and check whether the log are there or not.  A bit involved perhaps, but if having more cases where we want to quietly ignore some errors, something worth investing a bit of time now?  Never attempted to do this, but probably doable.  What do you think?

Reply:
It's up to you if you want to try redirecting `stdout` and searching for the output string, but I won't hold up the PR over it ;)

Reply:
I found the `gag` crate, https://docs.rs/gag/0.1.10/gag/, which does not work on windows, but the test can be disabled there, won't impact the ci build.

And more specifically `gag::BufferRedirect` which would provide what's needed, but not clear whether this would interfere with what rust does to eliminate output while running the test, even with `--nocapture` on.  Hoping to try this tonight.  If that create does not work, using `fork` with special output files might do it, but that's getting maybe over the top.

Reply:
`gag` works, but only when running the test with `--nocapture`, which I suspected, and in fact this is listed in the limitations.  So I don't know whether you'd find this acceptable or not.

An other idea, since the code is using its own macros for `warn`, would be to use a global variable, or thread local (only compiled when in test mode), to count the number of calls to `warn`, or allow the code to temporarily override the function used to output.  This might be better, no?


Also, any idea why the `tests::integrations::mempool_errors` test failed on ci with a port already in use.  Appears to work fine locally.